rhizome/host/lib/slice_setup.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+require "fileutils"
+
+class SliceSetup
+  def initialize(slice_name)
+    @slice_name = slice_name
+  end
+
+  def systemd_service
+    @systemd_service ||= File.join("/etc/systemd/system", @slice_name)
+  end
+
+  def prep(allowed_cpus)
+    install_systemd_unit(allowed_cpus)
+    start_systemd_unit
+  end
+
+  def purge
+    if File.exist? systemd_service
+      r "systemctl stop #{@slice_name.shellescape}"
+      FileUtils.rm_f(systemd_service)
+
+      r "systemctl daemon-reload"
+    end
+  end
+
+  def install_systemd_unit(allowed_cpus)
+    fail "BUG: unit name must not be empty" if @slice_name.empty?
+    fail "BUG: we cannot create system units" if @slice_name == "system.slice" || @slice_name == "user.slice"
+    fail "BUG: unit name cannot contain a dash" if @slice_name.include?("-")
```

Comment:
as long as you are checking inputs, check allowed_cpus for nil in particular, since that will expand into a blank in the unit file and IIRC in the unit file such a blank expansion means "clear that setting" (i.e. it could be relatively silent kind of "oops")

Reply:
done